diff --git a/node_modules/react-native-web/dist/cjs/exports/Image/index.js b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
index 36a3f8d..39811ed 100644
--- a/node_modules/react-native-web/dist/cjs/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
@@ -315,6 +315,7 @@ function (_React$Component) {
         accessibilityState = _this$props3.accessibilityState,
         accessible = _this$props3.accessible,
         blurRadius = _this$props3.blurRadius,
+        className = _this$props3.className,
         defaultSource = _this$props3.defaultSource,
         draggable = _this$props3.draggable,
         importantForAccessibility = _this$props3.importantForAccessibility,
@@ -385,6 +386,7 @@ function (_React$Component) {
       accessibilityRole: accessibilityRole,
       accessibilityState: accessibilityState,
       accessible: accessible,
+      className: className,
       importantForAccessibility: importantForAccessibility,
       nativeID: nativeID,
       onLayout: this._createLayoutHandler(finalResizeMode),
diff --git a/node_modules/react-native-web/dist/cjs/exports/ScrollView/ScrollViewBase.js b/node_modules/react-native-web/dist/cjs/exports/ScrollView/ScrollViewBase.js
index f7a5343..ee63e58 100644
--- a/node_modules/react-native-web/dist/cjs/exports/ScrollView/ScrollViewBase.js
+++ b/node_modules/react-native-web/dist/cjs/exports/ScrollView/ScrollViewBase.js
@@ -127,6 +127,7 @@ function (_React$Component) {
         accessibilityRole = _this$props.accessibilityRole,
         accessibilityState = _this$props.accessibilityState,
         children = _this$props.children,
+        className = _this$props.className,
         importantForAccessibility = _this$props.importantForAccessibility,
         nativeID = _this$props.nativeID,
         onLayout = _this$props.onLayout,
@@ -144,6 +145,7 @@ function (_React$Component) {
       accessibilityRole: accessibilityRole,
       accessibilityState: accessibilityState,
       children: children,
+      className: className,
       importantForAccessibility: importantForAccessibility,
       nativeID: nativeID,
       onLayout: onLayout,
diff --git a/node_modules/react-native-web/dist/cjs/exports/Text/index.js b/node_modules/react-native-web/dist/cjs/exports/Text/index.js
index 473ffbe..a02e660 100644
--- a/node_modules/react-native-web/dist/cjs/exports/Text/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/Text/index.js
@@ -50,7 +50,7 @@ function (_React$Component) {
       supportedProps.onKeyDown = this._createEnterHandler(onPress);
     }
 
-    supportedProps.classList = [classes.text, hasTextAncestor === true && classes.textHasAncestor, numberOfLines === 1 && classes.textOneLine, numberOfLines != null && numberOfLines > 1 && classes.textMultiLine]; // allow browsers to automatically infer the language writing direction
+    supportedProps.classList = [_this$props.className, classes.text, hasTextAncestor === true && classes.textHasAncestor, numberOfLines === 1 && classes.textOneLine, numberOfLines != null && numberOfLines > 1 && classes.textMultiLine]; // allow browsers to automatically infer the language writing direction
 
     supportedProps.dir = dir !== undefined ? dir : 'auto';
     supportedProps.ref = forwardedRef;
diff --git a/node_modules/react-native-web/dist/cjs/exports/View/filterSupportedProps.js b/node_modules/react-native-web/dist/cjs/exports/View/filterSupportedProps.js
index 6b33d2c..46de995 100644
--- a/node_modules/react-native-web/dist/cjs/exports/View/filterSupportedProps.js
+++ b/node_modules/react-native-web/dist/cjs/exports/View/filterSupportedProps.js
@@ -19,6 +19,7 @@ var supportedProps = {
   accessibilityState: true,
   accessible: true,
   children: true,
+  className: true,
   disabled: true,
   importantForAccessibility: true,
   nativeID: true,
diff --git a/node_modules/react-native-web/dist/cjs/exports/View/index.js b/node_modules/react-native-web/dist/cjs/exports/View/index.js
index efe1676..75772fa 100644
--- a/node_modules/react-native-web/dist/cjs/exports/View/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/View/index.js
@@ -59,7 +59,7 @@ function (_React$Component) {
       });
     }
 
-    supportedProps.classList = [classes.view];
+    supportedProps.classList = [this.props.className, classes.view];
     supportedProps.ref = this.props.forwardedRef;
     supportedProps.style = _StyleSheet.default.compose(hasTextAncestor && styles.inline, this.props.style);
 
diff --git a/node_modules/react-native-web/dist/exports/Image/index.js b/node_modules/react-native-web/dist/exports/Image/index.js
index 8281e36..9682fc3 100644
--- a/node_modules/react-native-web/dist/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/exports/Image/index.js
@@ -296,6 +296,7 @@ function (_React$Component) {
         accessibilityRole = _this$props3.accessibilityRole,
         accessibilityState = _this$props3.accessibilityState,
         accessible = _this$props3.accessible,
+        className = _this$props3.className,
         blurRadius = _this$props3.blurRadius,
         defaultSource = _this$props3.defaultSource,
         draggable = _this$props3.draggable,
@@ -367,6 +368,7 @@ function (_React$Component) {
       accessibilityRole: accessibilityRole,
       accessibilityState: accessibilityState,
       accessible: accessible,
+      className: className,
       importantForAccessibility: importantForAccessibility,
       nativeID: nativeID,
       onLayout: this._createLayoutHandler(finalResizeMode),
diff --git a/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js b/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
index fb1155f..27743e9 100644
--- a/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
+++ b/node_modules/react-native-web/dist/exports/ScrollView/ScrollViewBase.js
@@ -125,6 +125,7 @@ function (_React$Component) {
         accessibilityRole = _this$props.accessibilityRole,
         accessibilityState = _this$props.accessibilityState,
         children = _this$props.children,
+        className = _this$props.className,
         importantForAccessibility = _this$props.importantForAccessibility,
         nativeID = _this$props.nativeID,
         onLayout = _this$props.onLayout,
@@ -142,6 +143,7 @@ function (_React$Component) {
       accessibilityRole: accessibilityRole,
       accessibilityState: accessibilityState,
       children: children,
+      className: className,
       importantForAccessibility: importantForAccessibility,
       nativeID: nativeID,
       onLayout: onLayout,
diff --git a/node_modules/react-native-web/dist/exports/Text/index.js b/node_modules/react-native-web/dist/exports/Text/index.js
index 9f08836..54745a3 100644
--- a/node_modules/react-native-web/dist/exports/Text/index.js
+++ b/node_modules/react-native-web/dist/exports/Text/index.js
@@ -45,7 +45,7 @@ function (_React$Component) {
       supportedProps.onKeyDown = this._createEnterHandler(onPress);
     }
 
-    supportedProps.classList = [classes.text, hasTextAncestor === true && classes.textHasAncestor, numberOfLines === 1 && classes.textOneLine, numberOfLines != null && numberOfLines > 1 && classes.textMultiLine]; // allow browsers to automatically infer the language writing direction
+    supportedProps.classList = [this.props.className, classes.text, hasTextAncestor === true && classes.textHasAncestor, numberOfLines === 1 && classes.textOneLine, numberOfLines != null && numberOfLines > 1 && classes.textMultiLine]; // allow browsers to automatically infer the language writing direction
 
     supportedProps.dir = dir !== undefined ? dir : 'auto';
     supportedProps.ref = forwardedRef;
diff --git a/node_modules/react-native-web/dist/exports/View/filterSupportedProps.js b/node_modules/react-native-web/dist/exports/View/filterSupportedProps.js
index a7a6b96..f3cb9b1 100644
--- a/node_modules/react-native-web/dist/exports/View/filterSupportedProps.js
+++ b/node_modules/react-native-web/dist/exports/View/filterSupportedProps.js
@@ -14,6 +14,7 @@ var supportedProps = {
   accessibilityState: true,
   accessible: true,
   children: true,
+  className: true,
   disabled: true,
   importantForAccessibility: true,
   nativeID: true,
diff --git a/node_modules/react-native-web/dist/exports/View/index.js b/node_modules/react-native-web/dist/exports/View/index.js
index 68072e7..2c186d5 100644
--- a/node_modules/react-native-web/dist/exports/View/index.js
+++ b/node_modules/react-native-web/dist/exports/View/index.js
@@ -54,7 +54,7 @@ function (_React$Component) {
       });
     }
 
-    supportedProps.classList = [classes.view];
+    supportedProps.classList = [this.props.className, classes.view];
     supportedProps.ref = this.props.forwardedRef;
     supportedProps.style = StyleSheet.compose(hasTextAncestor && styles.inline, this.props.style);
 
diff --git a/node_modules/react-native-web/src/exports/Image/index.js b/node_modules/react-native-web/src/exports/Image/index.js
index 10d20d8..4e533dd 100644
--- a/node_modules/react-native-web/src/exports/Image/index.js
+++ b/node_modules/react-native-web/src/exports/Image/index.js
@@ -199,6 +199,7 @@ class Image extends React.Component<ImageProps, State> {
       accessibilityState,
       accessible,
       blurRadius,
+      className,
       defaultSource,
       draggable,
       importantForAccessibility,
@@ -271,6 +272,7 @@ class Image extends React.Component<ImageProps, State> {
         accessibilityRole={accessibilityRole}
         accessibilityState={accessibilityState}
         accessible={accessible}
+        className={className}
         importantForAccessibility={importantForAccessibility}
         nativeID={nativeID}
         onLayout={this._createLayoutHandler(finalResizeMode)}
diff --git a/node_modules/react-native-web/src/exports/ScrollView/ScrollViewBase.js b/node_modules/react-native-web/src/exports/ScrollView/ScrollViewBase.js
index 07737de..5c99fab 100644
--- a/node_modules/react-native-web/src/exports/ScrollView/ScrollViewBase.js
+++ b/node_modules/react-native-web/src/exports/ScrollView/ScrollViewBase.js
@@ -81,6 +81,7 @@ export default class ScrollViewBase extends React.Component<Props> {
       accessibilityRole,
       accessibilityState,
       children,
+      className,
       importantForAccessibility,
       nativeID,
       onLayout,
@@ -101,6 +102,7 @@ export default class ScrollViewBase extends React.Component<Props> {
         accessibilityRole={accessibilityRole}
         accessibilityState={accessibilityState}
         children={children}
+        className={className}
         importantForAccessibility={importantForAccessibility}
         nativeID={nativeID}
         onLayout={onLayout}
diff --git a/node_modules/react-native-web/src/exports/Text/index.js b/node_modules/react-native-web/src/exports/Text/index.js
index 5f79e86..fe4ff14 100644
--- a/node_modules/react-native-web/src/exports/Text/index.js
+++ b/node_modules/react-native-web/src/exports/Text/index.js
@@ -34,6 +34,7 @@ class Text extends React.Component<TextProps> {
     }
 
     supportedProps.classList = [
+      this.props.className,
       classes.text,
       hasTextAncestor === true && classes.textHasAncestor,
       numberOfLines === 1 && classes.textOneLine,
diff --git a/node_modules/react-native-web/src/exports/View/filterSupportedProps.js b/node_modules/react-native-web/src/exports/View/filterSupportedProps.js
index 5c23416..3ebac8a 100644
--- a/node_modules/react-native-web/src/exports/View/filterSupportedProps.js
+++ b/node_modules/react-native-web/src/exports/View/filterSupportedProps.js
@@ -15,6 +15,7 @@ const supportedProps = {
   accessibilityState: true,
   accessible: true,
   children: true,
+  className: true,
   disabled: true,
   importantForAccessibility: true,
   nativeID: true,
diff --git a/node_modules/react-native-web/src/exports/View/index.js b/node_modules/react-native-web/src/exports/View/index.js
index 43902f6..e6f1437 100644
--- a/node_modules/react-native-web/src/exports/View/index.js
+++ b/node_modules/react-native-web/src/exports/View/index.js
@@ -49,7 +49,7 @@ class View extends React.Component<ViewProps> {
       });
     }
 
-    supportedProps.classList = [classes.view];
+    supportedProps.classList = [this.props.className, classes.view];
     supportedProps.ref = this.props.forwardedRef;
     supportedProps.style = StyleSheet.compose(
       hasTextAncestor && styles.inline,
diff --git a/node_modules/react-native-web/src/exports/View/types.js b/node_modules/react-native-web/src/exports/View/types.js
index 853c252..2540b5a 100644
--- a/node_modules/react-native-web/src/exports/View/types.js
+++ b/node_modules/react-native-web/src/exports/View/types.js
@@ -85,6 +85,7 @@ export type ViewProps = {
   },
   accessible?: boolean,
   children?: any,
+  className?: string,
   forwardedRef?: any,
   hitSlop?: EdgeInsetsValue,
   importantForAccessibility?: 'auto' | 'yes' | 'no' | 'no-hide-descendants',
